geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/index/AbstractIndexMaintenanceIntegrationTest.java

```diff
@@ -81,6 +82,21 @@ public void setPdxStringFlagWithAStringShouldNotBeChangedAfterTheFirstCall() thr
     assertTrue(statusIndex.isIndexOnPdxKeys());
   }
 
+  @Test
```

Comment:
It will be nice to add integration tests on region with compression settings. 
1. By creating index without any entries in the region, adding entries to the region and executing query using the index.
2. By creating index on a region already having values.

Reply:
Yes...With this we will be exercising PdxString during index initialization, index maintenance and query evaluation.